docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -2008,6 +2011,159 @@ against those packages and its datatypes are not automatically converted.
+warnings with ``-Wno-upgrade-serialized-non-upgradeable-dependency``.
+
+.. _upgrade_dont_upload_daml_script:
+
+Avoid Depending on Daml Script Packages
+---------------------------------------
+
+We recommend only depending on Daml Script packages such as ``daml-script-lts``
+in dedicated packages for running tests written in Daml Script. These packages
+should not be part of your model and should not be uploaded to the ledger.
+
+Daml Script packages are not guaranteed to be upgradeable across SDK versions.
+For this reason, if you depend on a Daml Script datatype in a serializable
+position (e.g. the field of a template), you may find your package will rely on
+a Daml Script package in a way that can neither be removed nor upgraded to the
+next SDK version. At this point, your package and any of its SCU upgrades would
+be stuck on that SDK version.
+
+For example, suppose you have a ``main`` package that depends on
+``daml-script-lts`` from SDK 2.10.0.
+
+.. code:: yaml
+
+  version: 2.10.0
+  name: main
+  ...
+  dependencies:
+  - daml-prim
+  - daml-stdlib
+  - daml-script-lts
+  build-options:
+  - --target=1.17
+
+.. code:: daml
+
+  module Main where
+
+  import qualified Daml.Script
+
+  data MyData = MyData
+    { field1 : Daml.Script.PartyIdHint
+    }
+
+Because ``MyData`` is a serializable datatype, any changes to it must be valid
+upgrade changes (e.g. adding a field) for the ``main`` package itself to be
+validly upgraded. If SDK 2.10.X were to introduce a change to
+``daml-script-lts`` that is not a valid upgrade of the ``daml-script-lts`` in
+SDK 2.10.0, then ``field1`` will not be upgradeable to the next version of the
+SDK, nor can the field be dropped because the field is used in SCU checks.
+
+**Note**: While it is unlikely that an SDK update will break ``daml-script-lts``
+for upgrades, we still strongly recommend against it. In Daml 3.x, Canton may
+disallow uploading ``daml-script-lts`` entirely.
+
+At that point, all future development on ``main``, including future upgrades,
+would be locked to SDK 2.10.0. To bump the SDK version, ``main`` would have to
+be migrated via a manual upgrade tool with downtime -- existing downtime upgrade
+techniques are listed :ref:`here <upgrades-index>`.
+
+For this reason, we strongly recommend against using Daml Script as an upgrade
+dependency for any package going on the ledger. Whenever building the ``main``
+package above, you can expect a warning:
+
+.. code:: bash
+
+  > daml build
+  ...
+  warning while type checking data type Main.MyData:
+    This package depends on a datatype <package ID>:Daml.Script.Stable:PartyIdHint from <package ID> (daml-script-lts-stable, 0.0.0) with LF version 1.17.
+    
+    It is not recommended that >= LF1.17 packages use datatypes from Daml Script, because those datatypes will not be upgradeable.
+    Upgrade this warning to an error -Werror=upgrade-serialized-daml-script
+    Disable this warning entirely with -Wno-upgrade-serialized-daml-script
```

Comment:
```suggestion
    We do not recommend using datatypes from Daml Script in >= LF1.17 packages, because those datatypes are not upgradeable.
    Upgrade this warning to an error -Werror=upgrade-serialized-daml-script
    Remove this warning with -Wno-upgrade-serialized-daml-script
```